lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -806,7 +806,17 @@ struct AIEObjectFifoStatefulTransformPass
 
           int unrollFactor =
               computeLCM(objFifoSizes); // also counts original loop body
-
+          // if loop iterations < unrollFactor, unroll the loop fully
+          if (forLoop.getSingleLowerBound() || forLoop.getSingleUpperBound() ||
+              forLoop.getSingleStep()) {
+            int64_t tripCount =
+                constantTripCount(*(forLoop.getSingleLowerBound()),
+                                  *(forLoop.getSingleUpperBound()),
+                                  *(forLoop.getSingleStep()))
+                    .value_or(0);
+            if (tripCount < unrollFactor)
+              unrollFactor = tripCount;
```

Comment:
Should the `if` on line 810 have `&&` conditions? If any of them are false, but one of them is true, then I think `constantTripCount` will fail and `unrollFactor` gets set to zero.

Reply:
That makes sense.